### ISSUE TITLE
Make this helpers work in acceptance

### ISSUE THREAD

**Issue by cibernox** [cibernox/ember-native-dom-helpers#24]
Placeholder issue so we keep an eye on this.

Investigate how `Ember.Test.promise` works.

**Comment by rwjblue** [comment]
I think the best path forward for this is to refactor to our own `-private/wait` helper (instead of always using `ember-test-helpers/wait`), that looks _something_ like this:

```js
import Ember from 'ember';
import integrationTestWait from 'ember-test-helpers/wait';

function inAcceptanceTest() {
  // using the presence of global `wait` / `click` isn't the best
  // but it is essentially the only thing we have...
  return typeof self.wait === 'function' && typeof self.click === 'function';
}

export default function() {
  if (inAcceptancetest()) {
    /*
      entangle with global `Ember.Test.promise` chain
      to ensure things like:
      
      ```js
      import { find, click, fillIn } from 'ember-native-dom-helpers';
      test('fasdf', function(assert) {
        click('.foo');
        fillIn('.something-that-shows-up-after-foo-is-clicked');
        andThen(function() {
          assert.equal(find('.derp').textContents, 'hello world!');
        });
      });
      ```
    */
    Ember.Test.promise = Ember.RSVP.resolve(Ember.Test.promise)
      .then(() => self.wait());
  }

  return integrationTestWait();
}
```

**Comment by rwjblue** [comment]
I am going to rope @machty into some discussions here too, because I believe that we should also incorporate his learnings (which I just haven't had time to update in the RFC yet) that he described in https://github.com/emberjs/rfcs/pull/119#issuecomment-269373675.  

The resulting changes to the library are fairly small IMO, and will affect how we interop with current acceptance testing...

**Comment by pixelhandler** [comment]
@rwjblue I'm curious about the end game for using these native helpers work in acceptance tests, perhaps that is no globals, import only? That said will we just mask the global helpers in the time being? Or, would a recommendation be to import and prefix with `_` or some other indicator that these are the helpers you though you might find here?

**Comment by cibernox** [comment]
Closed vía #38
